File: python/2021/day 2/solution.py

def part1():
	with open('input.txt', 'r') as f:
		horizontal_pos = 0
		depth = 0

		data = [x.strip('\n') for x in f.readlines()]

	for ins in data:
		cmd = ins.split()[0]
		steps = int(ins.split()[1])
		if cmd == 'forward':
			horizontal_pos += steps

		elif cmd == 'down':
			depth += steps
		elif cmd == 'up':
			depth -= steps

	res = depth*horizontal_pos
	return res

def part2():
	with open('input.txt', 'r') as f:
		with open('input.txt', 'r') as f:
			horizontal_pos = 0
			depth = 0
			aim = 0
			data = [x.strip('\n') for x in f.readlines()]

	for ins in data:
		cmd = ins.split()[0]
		x = int(ins.split()[1])

		if cmd == 'forward':
			horizontal_pos += x
			depth += aim*x
		elif cmd == 'down':
			aim += x

		elif cmd == "up":
			aim -= x

	res = horizontal_pos * depth

	return res




if __name__ == "__main__":
	print(f"{part1.__name__} : {part1()}")
	print(f"{part2.__name__} : {part2()}")

File: c++/2021/day 2/solution.cpp
#include <iostream>
#include <fstream>
#include <sstream>



using namespace std;

int count_lines()
{
	ifstream input;
	input.open("input.txt");
	int line_count = 0;
	string word;
	while(getline(input, word))
	{
		line_count++;
	}
	return line_count;
}

stringstream getInput()
{
	ifstream input;
	input.open("input.txt");
	stringstream inp;
	string line;
	for (int i=0;i<count_lines();i++)
	{
		getline(input,line);
		inp << line << '\n';
	}
	for (int i=0;i<count_lines();i++)
	{
		getline(inp, line);
		cout << line << endl;
	}
	return inp;
}
int main()
{

	int line_count = count_lines();	

	stringstream stream = getInput();
	// ifstream input;
	// input.open("input.txt");
	// int pos[2] = {0,0};
	// for (int i=0; i<line_count; i++)
	// {
	// 	string cmd;
	// 	int steps;
			
	// 	input >> cmd >> steps;

	// 	if (cmd == "forward")
	// 	{
	// 		pos[0] += steps;
	// 	}
	// 	else if (cmd == "up")
	// 	{
	// 		pos[1] -= steps;
	// 	}
	// 	else if (cmd == "down")
	// 	{
	// 		pos[1] += steps;
	// 	}
	// }
	// int res = 1;
	// for (int elem : pos)
	// {
	// 	res *= elem;
	// }
	// cout << res;
}